src/datasets/types.ts

```diff
@@ -0,0 +1,50 @@
+import { DataCube } from 'src/models/data-cube/data-cube.model';
+
+export interface Dataset {
+  charts: ChartMetaType[];
+  dataCube: DataCube;
+}
+
+export type ChartMetaType = LineChartMeta | TabbedChartsMeta;
+
+export type XYChartData = LineChartData;
+
+export interface Datum<T, U> {
+  x: T;
+  y: U;
+}
+
+export interface ChartMeta<Type, DataGeneratorType> {
+  type: Type;
+  fetchs: DataGeneratorType;
+  title: string;
+  xlabel?: string;
+  ylabel?: string;
+}

```

Comment:
I think naming them `ChartMetaType` and `ChartMeta` is quite confusing since `Type` suffix is generally used to indicate a constructor type.

In addition, I think it's better to remove `Meta` suffix from both. It makes more sense for `Dataset.charts` to be of type `Chart[]` than `ChartMetaType[]`.

So my suggestion is to rename `ChartMetaType` to `Chart`, and `ChartMeta` to `BaseChart`.

Reply:
I have refactored the types based on your reviews and our discussion. Feel free to comment on the new commits.